Makefile

```diff
@@ -68,6 +69,7 @@ THESEUS_CARGO           := $(ROOT_DIR)/tools/theseus_cargo
 THESEUS_CARGO_BIN       := $(THESEUS_CARGO)/bin/theseus_cargo
 EXTRA_FILES             := $(ROOT_DIR)/extra_files
 LIMINE_DIR              := $(ROOT_DIR)/limine-prebuilt
+RUST_SOURCE             := $(ROOT_DIR)/ports/rust
```

Comment:
technically this is a port, but I think that we probably should move it into a top-level dir like `/rust` since it's so wildly different in nature than other port crates (which can typically just be built normally as regular crates)

Reply:
That also makes it easier to discover and document.